### PR TITLE
tests and implementation of copying a dll resource to a tempFile

### PR DESCRIPTION
**Pull request by AhmetCanSolak**
provided method will be heavily used in projects which depend on `coremem`. Developed with in a TDD manner. You can find the tests and the implementation. Also updated `build.gradle` and `.gitignore`files accordingly. Please provide comment/review to merge.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -85,6 +85,9 @@ dependencies
 	// BridJ:
 	compile 'com.nativelibs4java:bridj:0.7.0'
 
+	// Commons:
+	compile ('commons-io:commons-io:2.4')                             {transitive= true}
```
Comment (royerloic) [inline build.gradle]: Let's not get this huge dependency just for one method. Copy paste the function and adapt it for our needs instead...

File: src/main/java/coremem/util/ResourceHandler.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        File lFile = File.createTempFile("coremem",".dll");
+        lFile.deleteOnExit();
+
+        try (
+                InputStream an = getClass().getResourceAsStream(pRelativePathToFileInJar);
+                OutputStream out = new BufferedOutputStream(new FileOutputStream(lFile))
+        )
+        {
+            byte[] buffer = new byte[4096];
+            for (;;) {
```
Comment (royerloic) [inline src/main/java/coremem/util/ResourceHandler.java]: Are you sure you have to do this yourself? check online for less low-level approach that does not use dependencies..

File: src/main/java/coremem/util/ResourceHandler.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+    /**
+     * Method to copy a .dll resource from a jar
+     */
+    public File copyDLLfromJarToTempFile(String pRelativePathToFileInJar) throws IOException {
+        File lFile = File.createTempFile("coremem",".dll");
```
Comment (royerloic) [inline src/main/java/coremem/util/ResourceHandler.java]: the file name should also depend on file name of the dll, just get pRelativePath... into a File and get the filename from it...

File: build.gradle (hunk 1 above)
Comment (AhmetCanSolak) [reply build.gradle]: Okay. Indeed that one method was a really nice one but i will be coming with my best on this

File: src/main/java/coremem/util/ResourceHandler.java (hunk 3 above)
Comment (AhmetCanSolak) [reply src/main/java/coremem/util/ResourceHandler.java]: That name is just for prefix. But still you are right i think, better to distinguish prefixes too. Will be updating that.

**Comment by AhmetCanSolak** [comment]
Updated filename extraction part and added the loadResource directly functionality. One more commit will come for function naming conventions but you can provide review, next commit will not involve anything other than naming conventions.

**Comment by royerloic** [comment]
travis ci is failing, once it passes we will merge

**Comment by AhmetCanSolak** [comment]
> travis ci is failing, once it passes we will merge

https://travis-ci.org/ClearControl/coremem/jobs/487161424

Here it is. It is about `spotless.gradle` which is coming from master gradle repo.

**Pull request merged**